src/transport/http2.rs

```diff
@@ -383,84 +515,71 @@ impl fmt::Display for OpenBiError {
 impl std::error::Error for OpenBiError {}
 
 /// Future returned by `Channel::open_bi`.
+#[allow(clippy::type_complexity)]
 #[pin_project]
-pub struct OpenBiFuture<'a, In, Out>(
-    Option<Result<(ResponseFuture, flume::Sender<Out>), OpenBiError>>,
-    PhantomData<&'a In>,
-);
+pub struct OpenBiFuture<'a, In, Out> {
+    chan: Option<
+        Result<
+            (
+                ResponseFuture,
+                flume::Sender<io::Result<Bytes>>,
+                Arc<ChannelConfig>,
```

Comment:
Does this really need to return the config out?  If you really want to know the config shouldn't you ask the channel directly instead of getting it every time you open a bi-directional stream?

Reply:
You need the config to do this check.

Reply:
i'm not convinced yet, but let's see how this pans out